packages/terra-application-layout/src/ApplicationLayout.jsx

```diff
@@ -0,0 +1,244 @@
+          nameConfig,
+          utilityConfig,
+          extensions,
+        };
+        componentConfig.props = componentProps;
+        componentConfig.componentClass = ApplicationMenuWrapper;
+
+        menuComponentConfig[size] = componentConfig;
+      });
+
+      menuConfig.component = menuComponentConfig;
+      config[menuKey] = menuConfig;
+    });
+
+    return config;
+  }
+
+  /**
+   * Builds and returns the routing configuration object for the ApplicationLayout by injecting the RoutingMenu for top navigation
+   * and ApplicationMenuWrapper's as necessary.
+   */
+  static buildRoutingConfig(props) {
+    const { routingConfig } = props;
+
+    const updatedConfig = Object.assign({}, routingConfig, {
+      menu: ApplicationLayout.buildApplicationMenus(props, Object.assign({}, routingConfig.menu, ApplicationLayout.buildNavigationMenuConfig(props))),
+    });
+
+    return updatedConfig;
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      applicationLayoutRoutingConfig: ApplicationLayout.buildRoutingConfig(this.props),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.nameConfig !== nextProps.nameConfig ||
+        this.props.utilityConfig !== nextProps.utilityConfig ||
+        this.props.routingConfig !== nextProps.routingConfig ||
+        this.props.navigationItems !== nextProps.navigationItems ||
+        this.props.indexPath !== nextProps.indexPath) {
+      this.setState({
+        applicationLayoutRoutingConfig: ApplicationLayout.buildRoutingConfig(nextProps),
+      });
+    }
+  }
+
+  render() {
+    const { app, nameConfig, utilityConfig, navigationAlignment, navigationItems, indexPath, extensions } = this.props;
+    const { applicationLayoutRoutingConfig } = this.state;
+
+    return (
+      <NavigationLayout
+        app={app}
+        config={applicationLayoutRoutingConfig}
+        header={(
+          <ApplicationHeader
+            nameConfig={nameConfig}
+            utilityConfig={utilityConfig}
+            extensions={extensions}
+            applicationLinks={{
+              alignment: navigationAlignment,
+              links: navigationItems ? navigationItems.map((route, index) => ({
+                id: `application-layout-tab-${index}`,
+                path: route.path,
+                text: route.text,
+              })) : undefined,
+            }}
+          />
+        )}
+        indexPath={indexPath}
+      />
+    );
+  }
+}
+
+ApplicationLayout.propTypes = propTypes;
+ApplicationLayout.defaultProps = defaultProps;
+
+/**
+ * The ApplicationLayout is wrapped with a ModalManager on export to provide modal functionality
+ * for utility presentation and content convenience.
+ */
+export default withModalManager(ApplicationLayout);
+
+export { RoutingMenu };
+
+const Utils = {
+  helpers: Helpers,
+  utilityHelpers: UtilityHelpers,
+  propTypes: ApplicationLayoutPropTypes,
+};
+export { Utils };
```

Comment:
nit: update to `export { RoutingMenu, Utils };`

Reply:
Thanks, I knew something was off with that